### DISCUSSION TITLE
Handling "Multiple Declarations of type" error

### DISCUSSION THREAD

**Discussion by hughjackson** [SystemRDL/systemrdl-compiler#136] [Q&A] [answered]
In my design I load register descriptions from a couple of ipxact files provided by the vendor. These both include an addressBlock with the same name (but declared within the memoryMap component).

Is it possible to import the files in a way that doesn't cause this issue (e.g. keep imported components within the top level memoryMap to prevent name conflict)?

**Comment by amykyta3** [comment, marked as answer]
Yeah I agree the latest implementation of the IP-XACT importer is not great in handling this. I ran into this issue recently as well. When I refactored it, I did not anticipate the variety of ways that different IP vendors would cause name collisions, even within their own IP.

I think the proper solution is to explicitly prefix every imported name with the component name, (as well as the memoryMap name for the addressBlock)
